liteipc.lua

```diff
@@ -1,115 +1,102 @@
           backslash_buffer = {}
-        else
-          if #backslash_buffer > 0 then
-            result = result .. table.concat(backslash_buffer)
-            backslash_buffer = {}
-          end
-          result = result .. c
         end
+        result = result .. c
       end
-      if need_quote then
-        result = result .. '"'
-      end
-      return result
     end
-  else
-    quote_shell = function (arg)
-      if #arg == 0 then
-        return "''"
-      end
-      local is_safe = arg:match("[^%%%+%-%./_:=@0-9a-zA-Z]") == nil
-      if is_safe then
-        return arg
-      else
-        return "'" .. arg:gsub("'", "'\"'\"'") .. "'"
-      end
+    if need_quote then
+      result = result .. '"'
     end
+    return result
   end
-
-  local function escape_args(args)
-    local result = {}
-    for _, arg in ipairs(args) do
-      table.insert(result, quote_shell(arg))
+else
+  quote_shell = function (arg)
+    if #arg == 0 then
+      return "''"
     end
-    return table.concat(result, ' ')
-  end
-
-  function liteipc.start_process(args, cwd)
-    -- this solution with pushd and popd feels fragile af but i don't know of
-    -- any better way to make popen behave like i need it to
-    if cwd ~= nil then os.execute("pushd "..quote_shell(cwd)) end
-    -- we need to redirect stderr to stdout
-    local escaped_command = escape_args(args).." 2>&1"
-    if os.getenv("LINTPLUS_LITEIPC_DEBUG_SYNC_MODE_ARGS") then
-      print(escaped_command)
+    local is_safe = arg:match("[^%%%+%-%./_:=@0-9a-zA-Z]") == nil
+    if is_safe then
+      return arg
+    else
+      return "'" .. arg:gsub("'", "'\"'\"'") .. "'"
     end
-    local proc = setmetatable({
-      popen = io.popen(escaped_command, 'r'),
-    }, Process)
-    if cwd ~= nil then os.execute("popd") end
-    return proc
   end
+end
 
-  function Process.poll(self, callback)
-    for line in self.popen:lines() do
-      callback(line)
-    end
-
-    local ok, exit = self.popen:close()
-    local code = ok and 1 or 0
-    return exit, code
+local function escape_args(args)
+  local result = {}
+  for _, arg in ipairs(args) do
+    table.insert(result, quote_shell(arg))
   end
-
-  return liteipc
-
+  return table.concat(result, ' ')
 end
 
--- this is a huge hack to get liteipc_native.so loading to work but it should be
--- functional under most circumstances. at least i hope so
-function liteipc_loader.async()
-  local dir_sep, path_sep, sub = package.config:match("(.-)\n(.-)\n(.-)\n")
-  local this_file = debug.getinfo(1, 'S').source
-
-  local this_dir = fs.parent_directory(this_file):match("[@=](.*)")
-  package.cpath =
-    package.cpath .. path_sep .. this_dir .. dir_sep .. sub .. ".so"
+function liteipc.start_process(args, cwd)
+  local escaped_command = escape_args(args)
+  if os.getenv("LINTPLUS_LITEIPC_DEBUG_SYNC_MODE_ARGS") then
+    print(escaped_command)
+  end
+  local proc = setmetatable({
+    popen = process.start(args)
```

Comment:
Does Lite XL's subprocess module not have a way of opening the command with arguments passed explicitly as a table? I do not like the argument escaping solution, it's platform dependent and can cause security issues if not used properly.

Reply:
If you take a closer look you will notice that process.start is taking the args table intact, i left the escaping just for the debugging print, but I could remove the debugging part if you want.

Reply:
Oh that's right. The fact you left the debugging code in threw me off. It would be nice if you could remove that, then this PR can be merged.

Reply:
Removed the arguments debugging stuff